src/domain/service/note.ts

```diff
@@ -195,6 +195,43 @@ export default class NoteService {
     };
   }
 
+  /**
+   * Create note relation
+   * @param noteId - id of the current note
+   * @param parentPublicId - id of the parent note
+   */
+  public async createNoteRelation(noteId: NoteInternalId, parentPublicId: NotePublicId): Promise<boolean> {
+    const currenParentNote = await this.noteRelationsRepository.getParentNoteIdByNoteId(noteId);
+
+    /**
+     * Check if the note already has a parent
+     */
+    if (currenParentNote !== null) {
+      throw new DomainError(`Note already has parent note`);
+    }
+
+    const parentNote = await this.noteRepository.getNoteByPublicId(parentPublicId);
+
+    if (parentNote === null) {
+      throw new DomainError(`Incorrect parent note`);
+    }
+
+    let parentNoteId: number | null = parentNote.id;
```

Comment:
How existing parentNote can not have id?

Reply:
because i will override this variable in loop, that checks for cyclic reference. In this case, it can be null when we get root note

Reply:
so we have a note, which we got by `getNoteByPublicId`, checked it for `null` value, and then `id` can be `null` in some case?

Reply:
as i remember `getNoteByPublicId` can not return note with empty id